storey/redis_driver.py

```diff
@@ -0,0 +1,482 @@
+        """
+        The key containing a list of aggregation data.
+
+        The `redis_key_prefix` string should be the Redis key prefix used to store
+        data for a Storey container, table path, and key combination.
+        """
+        return f"{redis_key_prefix}:{list_attribute_name}"
+
+    @classmethod
+    def _convert_python_obj_to_redis_value(cls, value):
+        if isinstance(value, datetime):
+            if pd.isnull(value):
+                return None
+            return f'{cls.DATETIME_FIELD_PREFIX}{value.timestamp()}'
+        elif isinstance(value, timedelta):
+            return f'{cls.TIMEDELTA_FIELD_PREFIX}{value.total_seconds()}'
+        else:
+            if isinstance(value, float):
+                if value == math.inf or value == -math.inf:
+                    # Use the shorter infinity form (inf, -inf) to save space.
+                    value = str(value)
+                elif value % 1 == 0:
+                    # Store whole numbers as integers to save space.
+                    value = int(value)
+            return json.dumps(value)
+
+    @classmethod
+    def _convert_python_obj_to_lua_value(cls, value):
+        if isinstance(value, datetime):
+            if pd.isnull(value):
+                return None
+            return f'"{cls.DATETIME_FIELD_PREFIX}{value.timestamp()}"'
+        elif isinstance(value, timedelta):
+            return f'"{cls.TIMEDELTA_FIELD_PREFIX}{value.total_seconds()}"'
+        elif isinstance(value, bool):
+            return f'"{json.dumps(value)}"'
+        else:
+            if isinstance(value, float):
+                if value == math.inf:
+                    value = "math.huge"
+                elif value == -math.inf:
+                    value = "-math.huge"
+                elif value % 1 == 0:
+                    # Store whole numbers as integers to save space.
+                    value = int(value)
+
+            return json.dumps(value)
+
+    @classmethod
+    def convert_redis_value_to_python_obj(cls, value):
+        if value is None:
+            return None
+
+        str_value = cls.convert_to_str(value)
+
+        value = str_value.lower()
+        if value.startswith(cls.TIMEDELTA_FIELD_PREFIX):
+            ret = timedelta(seconds=float(value.split(":")[1]))
+        elif value.startswith(cls.DATETIME_FIELD_PREFIX):
+            ret = datetime.fromtimestamp(float(value.split(":")[1]), tz=timezone.utc)
+        elif value == "inf":
+            ret = math.inf
+        elif value == "-inf":
+            ret = -math.inf
+        elif value == "true":
+            ret = True
+        elif value == "false":
+            ret = False
+        else:
+            # if value is a number, convert type to int / float
+            if value[0] in ('-', '+'):
+                value = value[1:]
+            if value.replace('.', '', 1).isdigit():
+                ret = int(value) if int(float(value)) == float(value) else float(value)
```

Comment:
Am I missing something, or we just ignore the sign here?

Reply:
good catch - fixed